JunList.java

```diff
@@ -1,145 +1,25 @@
-import java.util.*;
-
 public class JunList {
     private int[] list;
     private int trackListPosition;
 
-    public static void main(String[] args) {
-        System.out.println("Hello. You are now in JunWorld where you can create a JunList.");
-        System.out.println("Let's start off with instruction specifications...\n");
-        instructionSpecifications();
-
-        Scanner userInput = new Scanner(System.in);
-
-        System.out.println("You can now input specifications for your starting list.");
-        System.out.println("Please give a number between 0 and 100 for number of values in starting list.");
-
-        int memory = userInput.nextInt(); // user memory input
-
-        //logic to determine if array can be made from user memory input
-        if (memory > 100 || memory < 0) {
-            System.out.println("Read the previous instruction head ass");
-            System.exit(0);
-        } else if (memory == 0) {
-            JunList userList = new JunList();
-            System.out.println("Thanks. Your starter list is: " + Arrays.toString(userList.list));
-            System.out.println("Dive in and start manipulating your list!");
-            userList.manipulateList(); //while loop to manipulate list however user decides
-        } else {
-            JunList userList = new JunList(memory);
-            System.out.println("Thanks. Your starter list is: " + Arrays.toString(userList.list));
-            System.out.println("Dive in and start manipulating your list!");
-            userList.manipulateList(); //while loop to manipulate list however user decides
-        }
-    }
-
     public JunList() {
         //constructor that takes no parameter but initializes an empty list
         trackListPosition = 0;
-        list = new int[1]; // prints null
+        list = new int[1];
     }
 
-    public JunList(int memory) {
-        //constructor that takes parameter argument
-        Scanner userInput = new Scanner(System.in);
-
-        trackListPosition = memory;
-        list = new int[memory * 2];
-
-        for (int i = 0; i < memory; i++) {
-            System.out.println("Thank you. Now please input each value in your array starting from index " + i);
-            list[i] = userInput.nextInt();
-        }
+    /*
+    Take in an array then use that to define the list
+     */
+    public JunList(int[] initial) {

```

Comment:
You should take in an array and use it to create your initial list. To the dev using this code the whole memory part is hidden.